python/graphstorm/model/gnn.py

```diff
@@ -754,14 +759,28 @@ def init_optimizer(self, lr, sparse_optimizer_lr=None, weight_decay=0, lm_lr=Non
             sparse_optimizer_lr = lr
         if len(sparse_params) > 0:
             if self.use_wholegraph_sparse_emb():
-                # To use wholegraph sparse optimizer, optimizer needs to be created
-                # before sparse embeddings. Within attach_wg_optimizer, we materialize
-                # the WG distributed tensor and then attach the optimizer.
-                emb_optimizer = create_wholememory_optimizer("adam", {})
-                for param in sparse_params:
-                    assert isinstance(param, WholeGraphDistTensor) and param.use_wg_optimizer, \
-                        "Please create params (WholeGraph tensor) with use_wg_optimizer=True."
-                    param.attach_wg_optimizer(emb_optimizer)
+                if sparse_params[0].optimizer is not None:
```

Comment:
i wonder if we should hide wholegraph into its own components. for example, we will have a wholegraph optimizer for sparse embeddings. so that we don't need to define if else everywhere.